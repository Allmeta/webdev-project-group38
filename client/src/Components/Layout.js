import React, { Component } from 'react'
import styled from 'styled-components'
import Header from './Header'
import { Card } from 'semantic-ui-react'
import MovieCard from './Card.js'
import axios from 'axios'

const WrapperFlex = styled.div`
  display: flex;
  background-color: #212733;
  flex-direction: column;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  color: white;
  height: 100vh;
  margin:0;
`
const StyledBody = styled.div`
  flex: 1;
  height: 100%;
  padding-top:20px;
  padding-left:10%;
  padding-right:10%;
`

class Layout extends Component {
  constructor (props) {
    super(props)
    this.state = {
      cards: []
    }
    this.image_base = 'https://image.tmdb.org/t/p/w500'
  }
  componentDidMount () {
    const instance = axios.create({
      baseURL: 'http://localhost:5025/api'
    })
    instance.get('/movies')
      .then((res) => {
        for (let i of res.data) {
          let str = '/genres/movie/id'.replace('id', i.movie_id)
          instance.get(str)
            .then((res2) => {
              i.genres = []
              for (let j of res2.data) {
                i.genres.push(j.name)
                // TODO set state
              }
            })
            .catch((err) => {
              console.log(err)
            })
        }
        this.setState({
          cards: res.data
        })
      })
      .catch((err) => {
        console.log(err)
      })
  }
  render () {
    return (
      <WrapperFlex>
        <Header />
        <StyledBody>
          <Card.Group>
            {this.state.cards.map((o) =>
              <MovieCard
                key={o.movie_id}
                img={this.image_base + o.poster_path}
                description={o.summary}
                title={o.title}
                popularity={Number(o.rating)}
                // genres={o.genres.join(' ')}
                genres='WIP'
              />
            )}
            {/* <MovieCard
              description="Lots of random latin words that have little meaning but to fill out descriptions like these."
              title="Lorem Ipsum"
              img={imeg}
              popularity={1234}
              genres="Action, Adventure, Thriller"/> */}

          </Card.Group>
        </StyledBody>
      </WrapperFlex >
    )
  }
}
export default Layout
