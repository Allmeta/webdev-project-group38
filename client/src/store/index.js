import { createStore, applyMiddleware, compose } from 'redux'
import { SearchFormReducer } from "../reducers/SearchFormReducer";
import thunk from 'redux-thunk'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(SearchFormReducer, composeEnhancers(
  applyMiddleware(thunk)
));

export default store;