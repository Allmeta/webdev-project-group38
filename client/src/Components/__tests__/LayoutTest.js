import React from 'react'
import Layout from '../Layout.js'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Provider } from 'react-redux'
import store from '../../store/index.js'

it('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><Layout /></Provider>).toJSON()
  expect(tree).toMatchSnapshot()
})
