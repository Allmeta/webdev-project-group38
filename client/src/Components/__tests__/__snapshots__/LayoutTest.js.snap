// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  position: relative;
  padding: 0;
  padding-left: 10%;
  padding-right: 10%;
  margin: 0;
  background-color: #4553F4;
  font-family: Verdana,Geneva,Tahoma,sans-serif;
  color: white;
  text-align: center;
  height: 110px;
}

.c2 {
  width: 200px;
  float: left;
}

.c3 {
  float: left;
  position: absolute;
  width: 1000px;
  left: 40%;
  top: 30px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #212733;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-family: Verdana,Geneva,Tahoma,sans-serif;
  color: white;
  height: 100vh;
  margin: 0;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
  padding-top: 10%;
  padding-left: 10%;
  padding-right: 10%;
}

@media screen and (max-width:500px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div
  className="c0"
>
  <header
    className="ReactStickyHeader_root"
  >
    <div
      className="ReactStickyHeader_fixed"
    >
      <div
        className="Header_root"
      >
        <div
          className="c1"
        >
          <img
            className="c2"
            src="test-file-stub"
          />
          <div
            className="c3"
          >
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="fields"
              >
                <div
                  className="three wide field"
                >
                  <div
                    className="ui input"
                  >
                    <input
                      name="title"
                      onChange={[Function]}
                      placeholder="Title"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="three wide field"
                >
                  <div
                    className="ui input"
                  >
                    <input
                      name="genre"
                      onChange={[Function]}
                      placeholder="Genre"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="two wide field"
                >
                  <button
                    className="ui icon button"
                    onClick={[Function]}
                    role="button"
                    type="submit"
                  >
                    <i
                      aria-hidden="true"
                      className="search icon"
                    />
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ReactStickyHeader_midground"
      style={
        Object {
          "backgroundColor": undefined,
          "backgroundImage": undefined,
          "height": null,
          "top": null,
        }
      }
    />
    <div
      className="ReactStickyHeader_background-bg"
      style={
        Object {
          "backgroundColor": undefined,
          "backgroundImage": undefined,
        }
      }
    />
    <div
      className="ReactStickyHeader_foreground"
      style={
        Object {
          "backgroundColor": undefined,
          "backgroundImage": undefined,
          "opacity": 1,
        }
      }
    />
    <section />
  </header>
  <div
    className="c4"
  >
    <div
      className="ui centered cards"
    />
  </div>
</div>
`;
